### PR TITLE
Add: zlib.0.7

### PR DESCRIPTION
**Pull request by madroach**
Fixing some compile time warnings / errors

### CODE REVIEW

File: packages/zlib/zlib.0.7/opam [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+authors: "Christopher Zimmermann <christopher@gmerlin.de>"
+homepage: "https://github.com/madroach/ocaml-zlib"
+dev-repo: "git+https://github.com/madroach/ocaml-zlib.git"
+bug-reports: "https://github.com/madroach/ocaml-zlib/issues"
+license: "ISC"
+build: [
+  ["dune" "build" "-p" "zlib"]
+  ["dune" "build" "@doc" "-p" "zlib"] {with-doc}
+]
+run-test: [["dune" "runtest" "-p" "zlib"] {ocaml:version >= "4.06"}]
```
Comment (mseri) [inline packages/zlib/zlib.0.7/opam:12]:
```suggestion
build: [
  ["dune" "build" "-p" name]
  ["dune" "build" "@doc" "-p" name] {with-doc}
  ["dune" "runtest" "-p" name] {ocaml:version >= "4.06" & with-test}
]
```
This is a bit more idiomatic nowadays

File: packages/zlib/zlib.0.7/opam [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+depends: [
+  "ocaml"
+  "dune"
```
Comment (mseri) [inline packages/zlib/zlib.0.7/opam:15]:
```suggestion
  "dune" {>= "1.1"}
```
According to the dune-project file

File: packages/zlib/zlib.0.7/opam [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+]
+synopsis: "Bindings to the zlib compression library"
+description: """
+providing deflate compression with or without zlib or gzip headers.
+
+This library uses bigarrays as buffers and can therefore release the OCaml
+runtime during (de)compression, allowing other OCaml threads to continue."""
+url {
+  src: "https://github.com/madroach/ocaml-zlib/archive/0.6.tar.gz"
+  checksum: "md5=f270d166bc3a1dde453e8c8715ad6df2"
```
Comment (mseri) [inline packages/zlib/zlib.0.7/opam:28]:
```suggestion
  checksum: [
    "sha256=4a003a1d54cf8cca2627393a1de8b6f307884401a0461e34977460d6ce049d94"
    "sha512=44ba069f1bca2cb1dc86639dd010aa166b0d72309824fcd3296c17c8a92eae16d56cd5ca42e74f0c2fe55e39e7729c20d4e712c4a3e6f50c29c2503a3124a642"
  ]
```
We are phasing out md5, please use sha256 or sha512

**Review by mseri — APPROVED** [review]
The rest of the failures are due to incorrect zlib versions I think

**Comment by mseri** [comment]
On 32 bit platforms it fails with
```

#=== ERROR while compiling zlib.0.7 ===========================================#
# context              2.2.0~beta3~dev | linux/x86_32 | ocaml-base-compiler.5.2.0 | pinned(https://github.com/madroach/ocaml-zlib/archive/0.6.tar.gz)
# path                 ~/.opam/5.2/.opam-switch/build/zlib.0.7
# command              ~/.opam/opam-init/hooks/sandbox.sh build dune runtest -p zlib
# exit-code            1
# env-file             ~/.opam/log/zlib-7-1c28fd.env
# output-file          ~/.opam/log/zlib-7-1c28fd.out
### output ###
# File "test/dune", line 2, characters 7-11:
# 2 |  (name test)
#            ^^^^
# (cd _build/default/test && ./test.exe)
# zlib
#   Random binary: Fatal error: exception Invalid_argument("Zlib.flate")
```
is this expected?

**Comment by rmonat** [comment]
@madroach?

**Comment by madroach** [comment]
No, this failure is not expected. I will need to setup a 32 bit ocaml installation to reproduce this. I suspect this is not a regression, but a pre-existing failure an 0.6, too. I'll have a chance to test this tomorrow.

**Pull request merged**